newIDE/app/src/InstructionOrExpression/EventsScope.flow.js

```diff
@@ -12,66 +12,97 @@ export type EventsScope = {|
   eventsFunctionsExtension?: gdEventsFunctionsExtension,
   eventsBasedBehavior?: ?gdEventsBasedBehavior,
   eventsBasedObject?: ?gdEventsBasedObject,
-  eventsFunction?: gdEventsFunction,
+  eventsFunction?: ?gdEventsFunction,
 |};
 
-export const getProjectScopedContainersFromScope = (
-  scope: EventsScope,
-  globalObjectsContainer: gdObjectsContainer,
-  objectsContainer: gdObjectsContainer
-): gdProjectScopedContainers => {
-  const {
-    project,
-    layout,
-    eventsFunctionsExtension,
-    eventsBasedBehavior,
-    eventsBasedObject,
-    eventsFunction,
-  } = scope;
-  if (layout) {
-    return gd.ProjectScopedContainers.makeNewProjectScopedContainersForProjectAndLayout(
-      project,
-      layout
-    );
+export class ProjectScopedContainers {
```

Comment:
Probably hyper risky for the reader to name a JS class with the exact same name as the `gd` class ;) 
I'm seeing `new ProjectScopedContainers` and panicking about this weird thing - and if I try to grep this in the codebase I now have a very bad time trying to distinguish between them.

This is also just "false" to say that this is a ProjectScopedContainers. It's something else by reading its content, even if it allows you to... generate/build/access to the proper ProjectScopedContainers

Rename this to `ProjectScopedContainersAccessorForEvents` :) 

(Important)

Reply:
ProjectScopedContainersAccessor if you want something shorter.